packages/text-annotator/src/model/w3c/W3CTextFormatAdapter.ts

```diff
@@ -6,10 +6,9 @@ import {
   parseW3CUser,
   serializeW3CBodies
 } from '@annotorious/core';
-import type { TextAnnotation, TextSelector } from '../core';
+import type { TextAnnotation, TextAnnotationTarget, TextSelector } from '../core';
 import type { W3CTextAnnotation, W3CTextSelector } from '../w3c';
-import { reviveTarget } from '../../state';
-import { getQuoteContext } from '../../utils';
+import { getQuoteContext, reviveSelector } from '../../utils';
```

Comment:
A note on this: I don't think this should happen here. The core should handle range reviving, not the crosswalk. In fact I believe that's already the case... it should be safe to omit reviving the range here, since it will happen when the annotation gets added to the store. (But didn't test yet.)

Reply:
Unfortunately, I think removing the revival from the `parseW3CTextAnnotation` might was a mistake. Because:
1. It breaks the typescript contract. The `parseW3CTextAnnotation` returns the `TextAnnotation`, where the `range` is a mandatory property. By removing the revival - the parsed selector won't have it 🤷🏻‍♂️ 
2. It breaks the adapter's tests that expect the `range` to present on the parsed selector

Reply:
I think that having 2 types might be a better path 🤔. Saying that the `range` is possibly undefined would be another TS "lie". As it cannot be the case within the Core. 

I would suggest adding not the `RevivedTextAnnotation`, but the `UnrevivedTextAnnotation` type instead. So we can leave all the existing code which uses the `TextAnnotation` in place. But precisely add usage of the `UnrevivedTextAnnotation` as a type for any annotations that come from outside of the lib

Reply:
The timing for this discussion turned out to be just right 😆 
I've realized that for our app we'll need a custom `reviveSelector` function that will process positions differently. So if we use the `reviveSelector` in the W3C adapter - it would require rewriting the adapter as well in the app). But w/o it - we'll just monkey patch the `addAnnotation` in the plugin and will execute our custom revived there

Reply:
Sounds good! I agree about your first point, 2 types being the better solution. However, following suggestion: would it make sense to have a `TextAnnotation` type where the range is optional, and a `RevivedTextAnnotation` where range is mandatory?

In the "edges" of the API, it's often the case that someone will hand in one or the other. (An unrevived annotation loaded from the backend, or a revived annotation that the pulled out from a "create" event or whatever.) The `revive` methods don't really care. They will check if a (non-collapsed) range is there, and revive the range if there isn't. That means the revive methods would take any `TextAnnotation` as input, and always return a `RevivedTextAnnotation` as a result. 

This way, in the core, we'd always work with RevivedTextAnnotations, while users of the API generally wouldn't need to care. (RevivedTextAnnotation is also a TextAnnotation, after all.)

Even in your case, you _could_ write your own custom adapter that returns a RevivedTextAnnotation, yet still use the core unchanged, right? (Or you could monkey patch it, too. In fact I've done the same for the PDF annotator. Although that started to feel even messier than the rest of the text annotator ;-)

Reply:
> would it make sense to have a TextAnnotation type where the range is optional, and a RevivedTextAnnotation where range is mandatory?

I would still recommend going the other way around. Because we already have soooo many places where the `TextAnnotation` is used. And renaming `TextAnnotation` to something else would lead to a pretty large hurdle. And we're not even talking about any code outside of the R6O library. So having another type with the optional `range` that will be used only on edges sound like a safer option to me

Reply:
I'll give it some thought. But TBH it just feels more sensible in terms of semantics to have the more generic name for the more generic "thing", while a more specialized name for the more narrowly defined type?

Reply:
Ok, I'm willing to go with the argument that `TextAnnotation` is the internal model, and that's supposed to come with range attached. But we'd definitely need a better name for the "other" thing then. "UnrevivedTextAnnotation" doesn't work, because the range would have to be optional, right? Something that conveys the idea that this could be both an internal annotation, as well as one that still needs to be revived.

Reply:
`RangelessAnnotation`? 😆 

> Something that conveys the idea that this could be both an internal annotation, as well as one that still needs to be revived.

That's a valid concern. I see how `RevivedTextAnnotation` works now - it conveys the idea of "narrowing" down the type via "reviving". And there's no ambiguity in what the `Revived...` means 🤔 
But it's not obvious to think  of a good inverse name that would have the same directness level